src/client/mesh_generator_thread.cpp

```diff
@@ -66,6 +77,10 @@ bool MeshUpdateQueue::addBlock(Map *map, v3s16 p, bool ack_block_to_server, bool
 
 	MutexAutoLock lock(m_mutex);
 
+
+	v3s16 original_block = p;
+	p.X &= ~1; p.Y &= ~1; p.Z &= ~1;
```

Comment:
Please put a comment here.
Without the text in the description I would not understood what is going on here.

Reply:
I've used a new variable for the mesh position instead of patching the existing one.